src/utils/openPopup.js

```diff
@@ -20,48 +20,60 @@ export const openPopup = (params) => {
   const container = dom.getContainer()
   const popup = dom.getPopup()
 
-  if (params.onBeforeOpen !== null && typeof params.onBeforeOpen === 'function') {
+  if (typeof params.onBeforeOpen === 'function') {
     params.onBeforeOpen(popup)
   }
 
-  if (params.animation) {
-    dom.addClass(popup, swalClasses.show)
-    dom.addClass(container, swalClasses.fade)
-  }
-  dom.show(popup)
+  addClasses({ container, popup })(params)
```

Comment:
What's the deal with functions returning functions? For me personally, it's quite hard to read such code, is it possible to achieve the same result with "simple" functions?

Robots (codeclimate) are happy with the code, but we should have humans in mind as well :)

Reply:
Yeah, of course. A higher-order function is not necessary here, I will change this.